src/lib/crypto/s2k.cpp

```diff
@@ -165,11 +169,13 @@ pgp_s2k_compute_iters(pgp_hash_alg_t alg, size_t desired_msec, size_t trial_msec
         trial_msec = DEFAULT_S2K_TUNE_MSEC;
     }
 
-    pgp_hash_t hash;
-    pgp_hash_create(&hash, alg);
+    if (!pgp_hash_create(&hash, alg)) {
+        RNP_LOG("failed to create hash object");
+        return 0;
+    }
 
-    const uint64_t start = get_timestamp_usec();
-    uint64_t       end = start;
```

Comment:
As above, it feels like a step backwards a bit, I think it's widely accepted that the best practice is to declare and initialize here.